docs/adr/ADR-004-contract-premiums.md

```diff
@@ -0,0 +1,40 @@
+# ADR 004 - Contract premiums
+
+## Status
+
+Already implemented.
+
+## Abstract
+
+Contract premiums allow smart contract developers to define a custom flat fee for interacting with their smart contract.
+
+## Context
```

Comment:
can we please add a section for `User Impact/Dapp Developer Impact` with query examples that users can use to get the premium fee? Ideally all new APIs/Queries should be included here IMHO